merlin/datasets/ecommerce/booking/dataset.py

```diff
@@ -0,0 +1,231 @@
+import os
+import urllib
+from glob import glob
+from pathlib import Path
+from typing import Optional, Tuple, Union
+
+from tqdm import tqdm
+
+import merlin.io
+from merlin.core.dispatch import get_lib
+from merlin.datasets import BASE_PATH
+from merlin.models.utils.example_utils import workflow_fit_transform
+from merlin.models.utils.nvt_utils import require_nvt
+from merlin.schema import Tags
+
+try:
+    import nvtabular as nvt
+
+    Workflow = nvt.Workflow
+except ImportError:
+    Workflow = None
+
+
+_FILES = ["ground_truth.csv", "test_set.csv", "train_set.csv"]
+_DATA_URL = "https://raw.githubusercontent.com/bookingcom/ml-dataset-mdt/main/"
+
+
+def get_booking(
+    path: Union[str, Path],
+    overwrite: bool = False,
+    transformed_name: str = "transformed",
+    nvt_workflow: Optional[Workflow] = None,
+    **kwargs,
+) -> Tuple[merlin.io.Dataset, merlin.io.Dataset]:
+    """_summary_
+
+    Args:
+        path (Union[str, Path]): _description_
+        overwrite (bool, optional): _description_. Defaults to False.
+        transformed_name (str, optional): _description_. Defaults to "transformed".
+        nvt_workflow (Optional[Workflow], optional): _description_. Defaults to None.
+
+    Returns:
+        Tuple[merlin.io.Dataset, merlin.io.Dataset]: _description_
+    """
+    require_nvt()
+
+    if path is None:
+        p = Path(BASE_PATH) / "booking"
+    else:
+        p = Path(path)
+
+    raw_path = p
+    if not raw_path.exists():
+        download_booking(p)
+
+    nvt_path = raw_path / transformed_name
+    train_path, valid_path = nvt_path / "train", nvt_path / "valid"
+    nvt_path_exists = train_path.exists() and valid_path.exists()
+    if not nvt_path_exists or overwrite:
+        transform_booking(raw_path, nvt_path, nvt_workflow=nvt_workflow, **kwargs)
+
+    train = merlin.io.Dataset(str(train_path), engine="parquet")
+    valid = merlin.io.Dataset(str(valid_path), engine="parquet")
+
+    return train, valid
+
+
+def download_booking(path: Path):
+    """_summary_
+
+    Args:
+        path (Path): _description_
+    """
+    path.mkdir(parents=True, exist_ok=True)
+
+    for file in _FILES:
+        local_filename = str(path / file)
+        url = os.path.join(_DATA_URL, file)
+        desc = f"downloading {os.path.basename(local_filename)}"
+        with tqdm(unit="B", unit_scale=True, desc=desc) as progress:
+
+            def report(chunk, chunksize, total):
+                if not progress.total:
+                    progress.reset(total=total)
+                progress.update(chunksize)
+
+            urllib.request.urlretrieve(url, local_filename, reporthook=report)
+
+    train = get_lib().read_csv(path / "train_set.csv")
```

Comment:
should we extract this block from the download function to a preprocessing function?